### ISSUE TITLE
Dialog: Remove autofocus attr from closeButton

### ISSUE THREAD

**Issue by oddvernes** [digdir/designsystem#4158]
- When a dialog opens, the first focusable element recieves focus by default anyway. The `closebutton` is hardcoded before children
- If I want to focus something else on open, such as a search input I should be able to add `autofocus` on it as per [the spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#accessibility), but the `closeButton` prevents it by taking `autofocus` first
